src/utils/knowledgeartifacts.js

```diff
@@ -55,6 +58,9 @@ function fetchEndpoint(url, planDefinition) {
     if (endpoint) {
       db.upsert(ENDPOINTS, endpoint, r => compareUrl(r.fullUrl, endpoint.fullUrl));
       debug(`Endpoint/${endpoint.id} saved to db`);
+
+      const baseUrl = endpoint.address.split('/$process-message')[0];
```

Comment:
So by default `getReferencedResource` will only return the resource but in this scenario we need the fullUrl as well. As such I had previously updated the method to take in an extra parameter called `returnFullUrl`. When this is true it returns the data structure `{ fullUrl: string, resource: Resource }` instead of just `Resource`. 

The fix was just to reference the `.resource` property in this `fetchEndpoint` function. This also fixes the problem where we were logging `Endpoint/undefined` by changing the line above to be `endpoint.resource.id`.